packages/ckeditor5-table/src/tablecellwidth/tablecellwidthediting.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * @license Copyright (c) 2003-2022, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module table/tablecellwidth/tablecellwidthediting
+ */
+
+import { Plugin } from 'ckeditor5/src/core';
+
+import { downcastAttributeToStyle, upcastStyleToAttribute } from './../converters/tableproperties';
+import TableEditing from './../tableediting';
+import TableCellWidthCommand from './commands/tablecellwidthcommand';
+import { getNormalizedDefaultProperties } from '../utils/table-properties';
+
+/**
+ * The table cell width editing feature.
+ *
+ * Introduces table cell model attribute and their conversion:
+ *
+ * - cell width: `tableCellWidth`
+ *
+ * It also registers a command used to manipulate the above attribute:
+ *
+ * - width: the `'tableCellWidth'` command
+ *
+ * @extends module:core/plugin~Plugin
+ */
+export default class TableCellWidthEditing extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	static get pluginName() {
+		return 'TableCellWidthEditing';
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	static get requires() {
+		return [ TableEditing ];
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	init() {
+		const editor = this.editor;
+		const schema = editor.model.schema;
+		const conversion = editor.conversion;
+
+		const defaultTableCellProperties = getNormalizedDefaultProperties(
+			editor.config.get( 'table.tableCellProperties.defaultProperties' )
+		);
+
+		enableProperty( schema, conversion, {
+			modelAttribute: 'tableCellWidth',
+			styleName: 'width',
+			defaultValue: defaultTableCellProperties.width
+		} );
+		editor.commands.add( 'tableCellWidth', new TableCellWidthCommand( editor, defaultTableCellProperties.width ) );
```

Comment:
Also, I just noticed that we have both a command and a model attribute called `tableCellWidth` :scream: it seems like a dangerous idea as it can be confusing, but AFAICS all (or most) of the table and table cell commands follow this pattern, so it looks like an intentional design :thinking: Still, we can't make such a change in isolation and it would be a big breaking change for all the integrators, so we'll have to live with it.

Reply:
Yes, i've seen that pattern and i followed with it.